rpc/provider/base.ts

```diff
@@ -71,3 +71,54 @@ export function nextIdFactory() {
   let i = 0
   return () => i++
 }
+
+export class ListenersContainer<
```

Comment:
The intent of this class is to replace the `ProviderConnection` class and use it in the `smoldot` provider too.
The `ProviderConnection` class has the `inner` and `cleanUp` props, but they are not used by the class methods, so it seems that they don't belong to this class.
After removing these props from `ProviderConnection`, this class has the responsibility of containing listeners for a single `WebSocket` connection.
The `ListenersContainer` is an improvement on the above that can contain listeners for many `WebSocket` connections.